app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala

```diff
@@ -16,36 +16,218 @@
 
 package uk.gov.hmrc.tariffclassificationfrontend.models
 
-import uk.gov.hmrc.tariffclassificationfrontend.models.Role.Role
+trait Permission {
+  def name: String
+}
+
+trait GlobalPermission extends Permission {
+  def appliesTo(operator: Operator): Boolean
+}
+
+trait CasePermission extends Permission {
+  def appliesTo(`case`: Case, operator: Operator): Boolean
+}
+
+object Permission {
+  private lazy val values: Set[Permission] = Set(
+    VIEW_MY_CASES,
+    VIEW_QUEUE_CASES,
+    VIEW_ASSIGNED_CASES,
+    SEARCH,
+    ADVANCED_SEARCH,
+    VIEW_CASES,
+    VIEW_CASE_ASSIGNEE,
+    VIEW_REPORTS,
+    CREATE_CASES,
+    ASSIGN_CASE,
+    RELEASE_CASE,
+    SUPPRESS_CASE,
+    REFER_CASE,
+    REOPEN_CASE,
+    REJECT_CASE,
+    SUSPEND_CASE,
+    COMPLETE_CASE,
+    REMOVE_ATTACHMENTS,
+    CANCEL_CASE,
+    ADD_NOTE,
+    ADD_ATTACHMENT,
+    KEYWORDS,
+    EDIT_LIABILITY,
+    EDIT_RULING,
+    APPEAL_CASE,
+    EXTENDED_USE,
+    MOVE_CASE_BACK_TO_QUEUE,
+    EDIT_SAMPLE
+  )
+
+  def from(string: String): Option[Permission] = values.find(_.name == string)
+  def applyingTo(operator: Operator): Set[Permission] = values
```

Comment:
All together is a great idea (Maybe more roles could mess but is not what this app is intended for)... not much to add here. Maybe I would define lazy structures to not iterate them again and again or some other little changes. But just a preference not a requirement 
`lazy val globalPermission =  values.filter(_.isInstanceOf[GlobalPermission]).map(_.asInstanceOf[GlobalPermission])`
`lazy val casePermissions = values.filter(_.isInstanceOf[CasePermission]).map(_.asInstanceOf[CasePermission])`
`def applyingTo(operator: Operator): Set[Permission] =  globalPermission.filter(_.appliesTo(operator)).map(_.asInstanceOf[Permission])`